matter_server/common/helpers/util.py

```diff
@@ -105,26 +86,20 @@ def parse_utc_timestamp(datetime_string: str) -> datetime:
     return datetime.fromisoformat(datetime_string.replace("Z", "+00:00"))
 
 
-def parse_value(
-    name: str, value: Any, value_type: Union[type[Any], str], default: Any = MISSING
-) -> Any:
-    """Try to parse a value from raw (json) data and type definitions."""
+def parse_value(name: str, value: Any, value_type: Any, default: Any = MISSING) -> Any:
+    """Try to parse a value from raw (json) data and type annotations."""
+
+    if isinstance(value_type, str):
+        value_type = get_type_hints(value_type, globals(), locals())
+
+    # always prefer classes that have a from_dict / FromDict
+    if isinstance(value, dict) and hasattr(value_type, "from_dict"):
+        return value_type.from_dict(value)
+    if isinstance(value, dict) and hasattr(value_type, "FromDict"):
+        return value_type.FromDict(value)
+
     if isinstance(value, dict) and "_type" in value:
```

Comment:
Since this 3 times checks `isinstance`, you can nest the 3 other checks. `isinstance`